l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -23,35 +23,38 @@ contract ExecutorFacet is Base, IExecutor {
     /// @dev Process one batch commit using the previous batch StoredBatchInfo
     /// @dev returns new batch StoredBatchInfo
     /// @notice Does not change storage
+    /// @custom:todo ADD BACK VIEW ONCE WE HAVE THE BLOB VERSIONED HASHES
     function _commitOneBatch(
         StoredBatchInfo memory _previousBatch,
         CommitBatchInfo calldata _newBatch,
         bytes32 _expectedSystemContractUpgradeTxHash
-    ) internal view returns (StoredBatchInfo memory) {
+    ) internal returns (StoredBatchInfo memory) {
         require(_newBatch.batchNumber == _previousBatch.batchNumber + 1, "f"); // only commit next batch
 
+        uint8 pubdataSource = uint8(bytes1(_newBatch.pubdataCommitments[0]));
+        require(pubdataSource == uint8(PubdataSource.Calldata) || pubdataSource == uint8(PubdataSource.Blob), "us");
+
         // Check that batch contain all meta information for L2 logs.
         // Get the chained hash of priority transaction hashes.
-        (
-            uint256 expectedNumberOfLayer1Txs,
-            bytes32 expectedPriorityOperationsHash,
-            bytes32 previousBatchHash,
-            bytes32 stateDiffHash,
-            bytes32 l2LogsTreeRoot,
-            uint256 packedBatchAndL2BlockTimestamp
-        ) = _processL2Logs(_newBatch, _expectedSystemContractUpgradeTxHash);
-
-        require(_previousBatch.batchHash == previousBatchHash, "l");
+        LogProcessingOutput memory logOutput = _processL2Logs(_newBatch, _expectedSystemContractUpgradeTxHash, PubdataSource(pubdataSource));
+
+        // ToDo: Adapt to handle dynamic number of blobs
+        bytes32[] memory blobCommitments;
+        if (pubdataSource == uint8(PubdataSource.Blob)) {
+            blobCommitments = _verifyBlobInformation(_newBatch.pubdataCommitments[1:], logOutput.blob1Hash, logOutput.blob2Hash);
+        }
```

Comment:
I'd personally like to see the require(logValue == keccak256(_newBatch.pubdataCommitments[1:]), "wp"); line around here..

so that we have all the verifications done in the same place.